tests/journal_lock.py/test_journal_lock.py

```diff
@@ -0,0 +1,349 @@
+"""Tests for journal_lock.py code."""
+# Tests:
+#  - Is file actually locked after obtain_lock().  Problem: We opened the
+#     file in a manner which means nothing else can open it.  Also I assume
+#     that the same process will either be allowed to lock it 'again' or
+#     overwrite the lock.
+#
+#     Expected failures if:
+#
+#      1. Lock already held (elsewhere).
+#      2. Can't open lock file 'w+'.
+#      3. Path to lock file doesn't exist.
+#      4. journaldir is None (default on Linux).
+#
+#  - Does release_lock() work?  Easier to test, if it's worked....
+#      1. return True if not locked.
+#      2. return True if locked, but successful unlock.
+#      3. return False otherwise.
+#
+#  - Can any string to pathlib.Path result in an invalid path for other
+#    operations?
+#
+#  - Not sure about testing JournalAlreadyLocked class.
```

Comment:
May want to throw this into the module docstring?

Reply:
I found the bit that I'd edited out in an earlier commit, so restored this and made it part of the module docstring.